### PR TITLE
 Upgrade from .NET 4.6.1 to 4.8.1

### PR DESCRIPTION
**Pull request by megahirt**
Upgrad to .Net Framework 4.8.1 from @ethnos-men

### CODE REVIEW

**Review by megahirt — COMMENTED** [review]
This is looking good to me!

File: src/SolidGui/packages.config
Comment (hahn-kev) [inline src/SolidGui/packages.config:1]: I actually think we can remove this, we're not using the old style nuget packages anymore. We can also remove the packages.config in the tests project, along with `src/packages/repositories.config`

**Review by hahn-kev — CHANGES_REQUESTED** [review]
Left a comment. I can make the change if we want.

I'm not sure if this is in scope, but we should add a package reference to `Microsoft.NET.Test.Sdk` in our test project, that will allow `dotnet test` to find the tests without requiring us to explicitly tell it which dll to test against.

**Comment by megahirt** [comment]
Hi @ethnos-men , no worries.  I fixed up your previous git branch and also created a new PR.  You should checkout the net481 branch that you created, then commit the change on that branch and push.  The PR will be updated automatically.

**Comment by megahirt** [comment]
image
If you are like me and prefer a nice visual GUI for managing git commits/branches, I recommend GitKraken.  It's free for public repos like this one.

**Comment by hahn-kev** [comment]
> @hahn-kev : Thank you for the feedback. I agree with the proposals. 
> 
> How to proceed? Need the suggested changes to be added to a commit in the net481 Featurebranch or how are they added to the pull request?
> 
> I am not familiar with the process in Github Pullrequests.
> 
> If it is not much work for you, you could just do the changes in my opinion. Otherwise I can also do them when you tell me where to commit (question above).

No problem. 

The typical PR workflow is a bit messed up in this case as Chris created this PR so you can't push changes so it. 

Normally though you would just be able to push changes to your branch for this PR and then we will be able to review those changes.

**Comment by megahirt** [comment]
Once @hahn-kev marks this as approved I am happy to have this merged in.  I will then create a new release and test out the new installer.

**Comment by hahn-kev** [comment]
@megahirt can we not build and test the installer before we merge?

**Comment by megahirt** [comment]
> @megahirt can we not build and test the installer before we merge?

technically, yes, we can do this by pushing a tag starting with v on the branch, and that should generate the installer / release.  Could also make the installer locally but would prefer to leave CI to do that.  Either way, it involves pushing a tag.  If I do that on a branch and I'm happy with it, I would delete the branch and sample tag, and then re-push a new tag for the real publish.  If I do it on master and I'm happy with it, then I'd just leave it where it was.  In either failure case, I would delete the tag and fix up whatever needs fixing.

**Comment by megahirt** [comment]
I will go ahead and push a new v1.0.2 tag to this branch and see how that goes.

**Review by josephmyers — APPROVED** [review]
I have given this a brief test, and things appear to work as expected. The only note I have is that the installer .exe name is "SolidInstaller-{AppVersion}.exe" -- this is likely because the artifact built on a branch.

**Comment by megahirt** [comment]
> I have given this a brief test, and things appear to work as expected. The only note I have is that the installer .exe name is "SolidInstaller-{AppVersion}.exe" -- this is likely because the artifact built on a branch.

Thanks @josephmyers for testing!

I'm pretty sure that the GHA itself is not quite right.  I remember this being an issue that I didn't figure out before moving on.

**Pull request merged**